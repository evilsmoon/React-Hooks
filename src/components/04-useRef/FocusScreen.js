import React, { useRef } from 'react';
import './ref.css'
export const FocusScreen = () => {
    

    const inputRef = useRef();

    const handelClick = ()=>{
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />
            <div className='form-group'>
                
            <input
                ref={inputRef}
                className='form-control'
                placeholder='you name' 
            />
            </div>
            <div className='form-group'>
                <button 
                    className='btn btn-danger'
                    onClick={handelClick}
                >
                    Focus
                </button>
            </div>
        </div>
    )
}
