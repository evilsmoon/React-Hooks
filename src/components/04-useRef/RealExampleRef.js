import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';
import './ref.css'
export const RealExampleRef = () => {
    
    const [show, setShow] = useState(false);
    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr/>
            {
             show && <MultipleCustomHooks/>

            }

            <div className='form-group'>
                <button
                    className='btn btn-info'
                    onClick={()=>{
                        setShow(!show)
                    }}
                >
                    Show/Hide
                </button>
            </div>
        </div>
    )
}
