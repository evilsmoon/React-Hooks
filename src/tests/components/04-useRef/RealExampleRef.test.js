import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('test RealExampleRef.js', () => {
    
    test('debe de mostrar correctamente', () => {
        
        const wrapper = shallow(<RealExampleRef/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);
    });

    test('debe de mostrar <MultipleCustomHooks/>', () => {
        
        const wrapper = shallow(<RealExampleRef/>);
        wrapper.find( 'button' ).simulate('click')
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
        
    })
    
    
})
