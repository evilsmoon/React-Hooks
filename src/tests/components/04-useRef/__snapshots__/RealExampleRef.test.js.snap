// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RealExampleRef.js debe de mostrar correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <div
    className="form-group"
  >
    <button
      className="btn btn-info"
      onClick={[Function]}
    >
      Show/Hide
    </button>
  </div>
</div>
`;
